waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingService.java

```diff
@@ -323,17 +317,7 @@ public PanopticOperationHandler getPanopticOperationHandler() {
 
       @Override
       public List<String> getAllDatabases() {
-        List<String> combined = new ArrayList<>();
-        try {
-          List<String> databases = primaryDatabasesCache.get(PRIMARY_KEY);
-          for (String database : databases) {
-            combined.add(primaryDatabaseMapping.transformOutboundDatabaseName(database));
-          }
-          combined.addAll(mappingsByDatabaseName.keySet());
-        } catch (ExecutionException e) {
-          LOG.warn("Can't fetch databases: {}", e.getCause().getMessage());
-        }
-        return combined;
+        return new ArrayList<>(mappingsByDatabaseName.keySet());
```

Comment:
I think there is some risk now that the you won't see a database added to the primary database (if it was added outside of WD). If a client reconnects it will see the up-to-date list. So probably fine.